src/main/java/org/sagebionetworks/bridge/hibernate/HibernateAdherenceRecordDao.java

```diff
@@ -70,7 +70,9 @@ protected QueryBuilder createQuery(AdherenceRecordsSearch search) {
         builder.append(BASE_QUERY);
         
         WhereClauseBuilder where = builder.startWhere(AND);
-        where.appendRequired("ar.userId = :userId", "userId", search.getUserId());
+        if (search.getUserId() != null) {
```

Comment:
Might be worth adding a test in HibernateAdherenceRecordDaoTest for a null userId.

Reply:
Added